llvm/lib/Target/SystemZ/SystemZISelLowering.cpp

```diff
@@ -6587,6 +6574,27 @@ SDValue SystemZTargetLowering::combineTruncateExtract(
   return SDValue();
 }
 
+// Replace ALoad with a new ATOMIC_LOAD with a result that is extended to VT
+// per ETy.
+static SDValue extendAtomicLoad(AtomicSDNode *ALoad, EVT VT, SelectionDAG &DAG,
+                                ISD::LoadExtType ETy) {
```

Comment:
Given that this patch adds extending atomic loads as a common-code feature, shouldn't now also common code attempt to match them in DAGCombine?  Why does this need to be platform-specific?   But I guess this can be looked at in a separate patch.